lib/buildpacks_registry_api/client.ex

```diff
@@ -1,64 +1,309 @@
+              _link:
+                "https://registry.buildpacks.io/api/v1/buildpacks/paketo-buildpacks/passenger/0.4.0",
+              version: "0.4.0"
+            }
+          ]
+        }
       ]
+  """
+  def search(term) do
+    params = %{matches: term}
+    response = get("/search", params)
+    Jason.decode!(response, keys: :atoms)
   end
 
-  @impl true
-  def process_response_body(body) do
-    case Jason.decode(body, keys: :atoms) do
-      {:ok, json} -> json
-      _ -> nil
-    end
-  end
+  @doc """
+  Get a list of buildpack versions for the namespaced buildpack. Returns a buildpack version map.
 
-  def search(term) do
-    params = %{matches: term}
+  ## Examples
+    iex> BuildpacksRegistryApi.Client.buildpack_version_list("heroku", "ruby")
+    %{
+      latest: %{
+        description: "",
+        homepage: "",
+        id: "f7c0dcc3-55ab-46b9-9a89-e35ff60aa7f8",
+        licenses: nil,
+        name: "ruby",
+        namespace: "heroku",
+        stacks: ["heroku-18", "heroku-20"],
+        version: "0.1.3"
+      },
+      versions: [
+        %{
+          _link: "https://registry.buildpacks.io//api/v1/buildpacks/heroku/ruby/0.1.3",
+          version: "0.1.3"
+        },
+        %{
+          _link: "https://registry.buildpacks.io//api/v1/buildpacks/heroku/ruby/0.1.2",
+          version: "0.1.2"
+        },
+        %{
+          _link: "https://registry.buildpacks.io//api/v1/buildpacks/heroku/ruby/0.1.1",
+          version: "0.1.1"
+        },
+        %{
+          _link: "https://registry.buildpacks.io//api/v1/buildpacks/heroku/ruby/0.1.0",
+          version: "0.1.0"
+        }
+      ]
+    }
+  """
+  @spec buildpack_version_list(binary(), binary()) :: map()
+  def buildpack_version_list(namespace, name) do
+    response = get("/buildpacks/#{namespace}/#{name}")
+    Jason.decode!(response, keys: :atoms)
+  end
 
-    {:ok, %HTTPoison.Response{body: body, status_code: 200}} =
-      __MODULE__.get("/search", [], params: params)
+  @doc """
+  Returns detailed information for the buildpack version.any()
 
-    body
+  ## Examples
+    iex> BuildpacksRegistryApi.Client.buildpack_version_info("heroku", "ruby", "0.1.3")
+    %{
+      addr:
+        "public.ecr.aws/heroku-buildpacks/heroku-ruby-buildpack@sha256:3eeb3773cdbd29d4fb0d578f7781fe8c525de73593480e2740b7143262e5bef5",
+      created_at: "2021-10-27T21:00:43.307Z",
+      description: "",
+      homepage: "",
+      id: "f7c0dcc3-55ab-46b9-9a89-e35ff60aa7f8",
+      licenses: nil,
+      name: "ruby",
+      namespace: "heroku",
+      stacks: ["heroku-18", "heroku-20"],
+      updated_at: "2022-03-17T21:20:26.981Z",
+      version: "0.1.3",
+      version_major: "0",
+      version_minor: "1",
+      version_patch: "3",
+      yanked: false
+    }
+  """
+  @spec buildpack_version_info(binary(), binary(), binary()) :: map()
+  def buildpack_version_info(namespace, name, version) do
+    response = get("/buildpacks/#{namespace}/#{name}/#{version}")
```

Comment:
These urls are magic strings duplicated between the clients, they could be moved to another module and invoked as functions to make maintenance easier.

Reply:
I think I'm going to punt on that and keep the duplication in favor of readability and locality for the URL values.  I expect the URLs will not update frequently and the risk of introducing a bug between the two client versions seems small.